test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtensionState.java

```diff
@@ -69,4 +65,28 @@ protected void setTestFailed(Throwable failure) {
     protected void doClose() throws IOException {
 
     }
+
+    private static final class ShutdownThread extends Thread {
+
+        public ShutdownThread(QuarkusTestExtensionState state) {
+            super(new ShutdownHook(state), "Quarkus Test Cleanup Shutdown Thread");
+        }
+    }
+
+    private static final class ShutdownHook implements Runnable {
+
+        private final QuarkusTestExtensionState state;
+
+        private ShutdownHook(QuarkusTestExtensionState state) {
+            this.state = state;
+        }
+
+        @Override
+        public void run() {
+            try {
+                state.close();
```

Comment:
I wonder if I would have set `state` to null here too, just to be safe.

Reply:
Yeah, that makes sense

Reply:
Done